app/react/edge/edge-stacks/utils.ts

```diff
@@ -4,10 +4,15 @@ import { EnvironmentType } from '@/react/portainer/environments/types';
 
 import { EditorType } from './types';
 
-export function getValidEditorTypes(endpointTypes: EnvironmentType[]) {
+export function getValidEditorTypes(
+  endpointTypes: EnvironmentType[],
+  allowKubeToSelectCompose?: boolean
+) {
   const right: Partial<Record<EnvironmentType, EditorType[]>> = {
     [EnvironmentType.EdgeAgentOnDocker]: [EditorType.Compose],
-    [EnvironmentType.EdgeAgentOnKubernetes]: [EditorType.Kubernetes],
+    [EnvironmentType.EdgeAgentOnKubernetes]: allowKubeToSelectCompose
```

Comment:
Missing comment from EE above this line.  Do you want that in too?

`// allow kube to view compose if it's an existing kube compose stack`